test/gcs/helper_container_test.go

```diff
@@ -0,0 +1,290 @@
+//go:build linux
+
+package gcs
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+
+	"github.com/containerd/containerd/namespaces"
+	ctrdoci "github.com/containerd/containerd/oci"
+	oci "github.com/opencontainers/runtime-spec/specs-go"
+
+	"github.com/Microsoft/hcsshim/internal/guest/prot"
+	"github.com/Microsoft/hcsshim/internal/guest/runtime"
+	"github.com/Microsoft/hcsshim/internal/guest/runtime/hcsv2"
+	"github.com/Microsoft/hcsshim/internal/guest/stdio"
+	"github.com/Microsoft/hcsshim/internal/guest/storage"
+	"github.com/Microsoft/hcsshim/internal/guest/storage/overlay"
+	"github.com/Microsoft/hcsshim/internal/guestpath"
+
+	testoci "github.com/Microsoft/hcsshim/test/internal/oci"
+)
+
+// todo: autogenerate/fuzz realistic specs
+
+//
+// testing helper functions for generic container management
+//
+
+const tailNull = "tail -f /dev/null"
+
+// will call unmountRootfs during cleanup
+func createStandaloneContainer(ctx context.Context, t testing.TB, host *hcsv2.Host, id string, extra ...ctrdoci.SpecOpts) *hcsv2.Container {
```

Comment:
Could we also add some comments on what this func expects? IOW something like "this function assumes the rootfs has not be mounted and will clean up any resources created in the event of an error"